From 1ab7b80da28a2656582638aae65cfea97b26c5c0 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Tue, 30 May 2017 00:02:52 +0530
Subject: [PATCH] display: Don't animate screen brightness when turning the screen on

Make the display wake from sleep faster by going directly to the target
brightness instead of ramping up to it.

diff --git a/services/core/java/com/android/server/display/RampAnimator.java b/services/core/java/com/android/server/display/RampAnimator.java
index d71269f9eae..5625e348009 100644
--- a/services/core/java/com/android/server/display/RampAnimator.java
+++ b/services/core/java/com/android/server/display/RampAnimator.java
@@ -59,7 +59,7 @@ public RampAnimator(T object, IntProperty<T> property) {
      */
     public boolean animateTo(int target, int rate) {
         // Immediately jump to the target the first time.
-        if (mFirstTime || rate <= 0) {
+        if (mFirstTime || rate <= 0 || mCurrentValue == 0) {
             if (mFirstTime || target != mCurrentValue) {
                 mFirstTime = false;
                 mRate = 0;
